scrapbook/models.py

```diff
@@ -323,23 +281,30 @@ def reglue(self, name, new_name=None, raise_on_missing=True):
                 )
         else:
             scrap = self.scraps[name]
+            if new_name:
+                scrap = scrap._replace(name=new_name)
             if scrap.data is not None:
-                data = {
-                    GLUE_PAYLOAD_FMT.format(encoder=scrap.encoder): scrap_to_payload(
-                        scrap
-                    )
-                }
-                metadata = {"scrapbook": dict(name=name)}
-                # Call raw display function
+                data, metadata = _prepare_ipy_data_format(
+                    scrap.name, scrap_to_payload(scrap), scrap.encoder
+                )
+                # A little non-sensical in this case, but when regluing unattached
```

Comment:
Perhaps: Skip saving data for later regluing and remove 'scrapbook' from keys, when unattached

Removing line 293 as well.